pytest/tests/loadtest/locust/common/social.py

```diff
@@ -0,0 +1,270 @@
+        }
+      }
+    }
+    ```
+    """
+    updates = values
+    updates["index"] = social_db_build_index_obj(index)
+    msg = {"data": {sender: updates}}
+    return json.dumps(msg)
+
+
+def social_follow_args(sender: str, follow_list: "list[str]"):
+    follow_map = {}
+    graph_map = {"follow": {}}
+    notify_map = {}
+    for user in follow_list:
+        follow_map[user] = ""
+        graph_map["follow"] = {"type": "follow", "accountId": user}
+        notify_map[user] = {"type": "follow"}
+
+    values = {
+        "graph": {
+            "follow": follow_map
+        },
+    }
+    index = {"graph": graph_map, "notify": notify_map}
+    return social_db_set_msg(sender, values, index)
+
+
+def social_post_args(sender: str, text: str):
+    values = {"post": {"main": json.dumps({"type": "md", "text": text})}}
+    index = {"post": {"main": {"type": "md"}}}
+    msg = social_db_set_msg(sender, values, index)
+    return msg
+
+
+class TestSocialDbSetMsg(unittest.TestCase):
+
+    def test_follow(self):
+        sender = "alice.near"
+        follow_list = ["bob.near"]
+        msg = social_follow_args(sender, follow_list)
+        parsed_msg = json.loads(msg)
+        expected_msg = {
+            "data": {
+                "alice.near": {
+                    "graph": {
+                        "follow": {
+                            "bob.near": ""
+                        }
+                    },
+                    "index": {
+                        "graph":
+                            "[{\"key\": \"follow\", \"value\": {\"type\": \"follow\", \"accountId\": \"bob.near\"}}]",
+                        "notify":
+                            "[{\"key\": \"bob.near\", \"value\": {\"type\": \"follow\"}}]"
+                    }
+                }
+            }
+        }
+        self.maxDiff = 2000  # print large diffs
+        self.assertEqual(parsed_msg, expected_msg)
+
+    def test_post(self):
+        sender = "alice.near"
+        text = "#Title\n\nbody"
+        msg = social_post_args(sender, text)
+        parsed_msg = json.loads(msg)
+        expected_msg = {
+            "data": {
+                "alice.near": {
+                    "post": {
+                        "main":
+                            "{\"type\": \"md\", \"text\": \"#Title\\n\\nbody\"}"
+                    },
+                    "index": {
+                        "post":
+                            "[{\"key\": \"main\", \"value\": {\"type\": \"md\"}}]"
+                    }
+                }
+            }
+        }
+        self.maxDiff = 2000  # print large diffs
+        self.assertEqual(parsed_msg, expected_msg)
+
+
+@events.init.add_listener
+def on_locust_init(environment, **kwargs):
+    # `master_funding_account` is the same on all runners, allowing to share a
+    # single instance of SocialDB in its `social` sub account
+    funding_account = environment.master_funding_account
```

Comment:
Similar to the issue you opened earlier, here I'm getting an error (and sometimes a similar error in `ft.py`):

```
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/locust/event.py", line 40, in fire
    handler(**kwargs)
  File "/Users/marcelo/nearcore/pytest/tests/loadtest/locust/common/social.py", line 237, in on_locust_init
    funding_account = environment.master_funding_account
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Environment' object has no attribute 'master_funding_account'
```

Not sure how you wanna workaround this, but it worked for me with:

```diff
diff --git a/pytest/tests/loadtest/locust/common/base.py b/pytest/tests/loadtest/locust/common/base.py
index e26e8109f..bc8b6c9ea 100644
--- a/pytest/tests/loadtest/locust/common/base.py
+++ b/pytest/tests/loadtest/locust/common/base.py
@@ -15,6 +15,7 @@ import multiprocessing
 import pathlib
 import sys
 import time
+import threading
 
 sys.path.append(str(pathlib.Path(__file__).resolve().parents[4] / 'lib'))
 
@@ -298,6 +299,7 @@ def evaluate_rpc_result(rpc_result):
             raise ReceiptError(outcome["status"], receipt["id"])
     return result
 
+BASE_INIT_DONE = threading.Event()
 
 # called once per process before user initialization
 @events.init.add_listener
@@ -340,6 +342,7 @@ def on_locust_init(environment, **kwargs):
 
     NearUser.funding_account = funding_account
     environment.master_funding_account = master_funding_account
+    BASE_INIT_DONE.set()
 
 
 # Add custom CLI args here, will be available in `environment.parsed_options`
diff --git a/pytest/tests/loadtest/locust/common/ft.py b/pytest/tests/loadtest/locust/common/ft.py
index 6fd78467f..c75957727 100644
--- a/pytest/tests/loadtest/locust/common/ft.py
+++ b/pytest/tests/loadtest/locust/common/ft.py
@@ -11,7 +11,7 @@ import key
 import transaction
 
 from account import TGAS
-from common.base import Account, CreateSubAccount, Deploy, NearUser, send_transaction, Transaction
+from common.base import Account, CreateSubAccount, Deploy, NearUser, send_transaction, Transaction, BASE_INIT_DONE
 
 
 class FTContract:
@@ -127,6 +127,7 @@ class InitFTAccount(Transaction):
 
 @events.init.add_listener
 def on_locust_init(environment, **kwargs):
+    BASE_INIT_DONE.wait()
     # Note: These setup requests are not tracked by locust because we use our own http session
     host, port = environment.host.split(":")
     node = cluster.RpcNode(host, port)
diff --git a/pytest/tests/loadtest/locust/common/social.py b/pytest/tests/loadtest/locust/common/social.py
index e472ccafa..4e319c670 100644
--- a/pytest/tests/loadtest/locust/common/social.py
+++ b/pytest/tests/loadtest/locust/common/social.py
@@ -11,7 +11,7 @@ import transaction
 from account import TGAS, NEAR_BASE
 import cluster
 import key
-from common.base import Account, CreateSubAccount, Deploy, NearUser, Transaction, send_transaction
+from common.base import Account, CreateSubAccount, Deploy, NearUser, Transaction, send_transaction, BASE_INIT_DONE
 from locust import events, runners
 from transaction import create_function_call_action
 
@@ -232,6 +232,7 @@ class TestSocialDbSetMsg(unittest.TestCase):
 
 @events.init.add_listener
 def on_locust_init(environment, **kwargs):
+    BASE_INIT_DONE.wait()
     # `master_funding_account` is the same on all runners, allowing to share a
     # single instance of SocialDB in its `social` sub account
     funding_account = environment.master_funding_account

```

Reply:
Yeah I think this fix would make sense. But I'd rather add it in a separate PR. Can you maybe create this PR for it? (Or maybe I will create one)

Also, I have never seen this error when waiting for workers and master to start up properly.